pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -194,6 +195,46 @@ func (am *Alertmanager) CompareAndSendConfiguration(ctx context.Context, config
 	return nil
 }
 
+func (am *Alertmanager) SendConfiguration(ctx context.Context, config *models.AlertConfiguration) error {
+	postableConfig, err := notifier.Load([]byte(config.AlertmanagerConfiguration))
+	if err != nil {
+		return err
+	}
+
+	// Decode and decrypt secure fields before sending the config.
+	for _, rcv := range postableConfig.AlertmanagerConfig.Receivers {
```

Comment:
nit: You could make this a method of `AlertmanagerConfig` and avoid having to define it here. Something like `postableConfig.AlertmanagerConfig.Descrypt(decrypter)`